### PR TITLE
Gutenberg: Display the reason why a block is unavailable

### PR DESCRIPTION
**Pull request by dnalves**
* Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4005
* Gutenberg PR: https://github.com/WordPress/gutenberg/pull/35021
* WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17293

To test: Follow instructions from: https://github.com/WordPress/gutenberg/pull/35021

## Regression Notes
1. Potential unintended areas of impact
The Gutenberg editor 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added the other types of blocks to the post.

3. What automated tests I added (or what prevented me from doing so)
4. None

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java [hunk 1]
```diff
@@ -518,6 +521,11 @@ public void onGotoCustomerSupportOptions() {
                         mEditorFragmentListener.onGotoCustomerSupportOptions();
                     }
                 },
+                new OnNativeNoticeListener() {
```
Comment (jd-alexander) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java]: @dnalves could this be called `OnNoticeListener`?

Comment (dnalves) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java]: Oh, my bad, I missed this! I'll update it right away

Comment (dnalves) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java]: Done! 😄 
https://github.com/wordpress-mobile/WordPress-Android/pull/15435/commits/9a99e20d0242602032fe3cd08d81ad37ede14f44

**Comment by jd-alexander** [comment]
@dnalves I am thinking of testing this but I just saw the comment `THIS STILL NEEDS TO POINT TO THE RIGHT GUTENBERG-MOBILE COMMIT` Let me know if you need any help with setting this up so I can do a final test.

**Comment by dnalves** [comment]
> @dnalves I am thinking of testing this but I just saw the comment `THIS STILL NEEDS TO POINT TO THE RIGHT GUTENBERG-MOBILE COMMIT` Let me know if you need any help with setting this up so I can do a final test.

Updated the branch to point to your PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15435/commits/29db9666d9382c56e66384ddc5f09f9ae8889b16

**Comment by jd-alexander** [comment]
WP Android PR to perform build https://github.com/wordpress-mobile/WordPress-Android/pull/15463

**Review by jd-alexander — APPROVED** [review]
⛔️Thanks for these changes @dnalves the code works as expected. **DO NOT MERGE**. I will help with merging these changes in after the `gutenberg-mobile` PR and `gutenberg PR` has been merged in. Great work here 🥳

**Comment by dnalves** [comment]
I believe this shouldn't be closed. I don't have permission to re-open it (and change the branch it points to).

cc @jd-alexander

**Comment by oguzkocer** [comment]
It looks like this got auto-closed when #15478 was merged. I certainly did not intentionally close it. I think this might have happened because that base branch is deleted, and for forks it doesn't seem to auto-update the base branch as it does for regular PRs.

I've restored the `gutenberg/after_1.64.0` branch and re-opened this PR. The base branch probably needs to be updated to `develop` and then we can delete the `gutenberg/after_1.64.0` branch again. I didn't want to update the base branch as well, just in case there is a different process in place.

Let me know if you need any help with this @jd-alexander.

**Comment by jd-alexander** [comment]
> It looks like this got auto-closed when #15478 was merged. I certainly did not intentionally close it. I think this might have happened because that base branch is deleted, and for forks it doesn't seem to auto-update the base branch as it does for regular PRs.
> 
> 
> 
> I've restored the `gutenberg/after_1.64.0` branch and re-opened this PR. The base branch probably needs to be updated to `develop` and then we can delete the `gutenberg/after_1.64.0` branch again. I didn't want to update the base branch as well, just in case there is a different process in place.
> 
> 
> 
> Let me know if you need any help with this @jd-alexander.

Agreed. The base branch needs to be updated to develop since the release process has wrapped up. @dnalves you can do that update so that we can do a final test and start the merging process.

**Comment by AliSoftware** [comment]
Moving this to milestone 18.8 since I'm doing to 18.7 code-freeze today and there are still conflicts.

**Pull request closed without merging**